packages/react-magma-dom/src/components/AlertBase/index.tsx

```diff
@@ -275,16 +276,39 @@ const StyledAlertInner = styled.div<AlertBaseProps>`
     `}
 `;
 
-const AlertContents = styled.div`
+const AlertContents = styled.div<{
+  additionalContent?: React.ReactNode;
+  isDismissible?: boolean;
+}>`
+  align-items: ${props => (props.additionalContent ? 'center' : '')};
   align-self: center;
   flex-grow: 1;
   padding: ${props => props.theme.spaceScale.spacing04} 0;
+  display: ${props => (props.additionalContent ? 'flex' : '')};
+  margin-right: ${props =>
+    props.additionalContent && !props.isDismissible
+      ? props.theme.spaceScale.spacing03
+      : ''};
+  * {
+    margin-left: ${props => (props.additionalContent ? '7px' : '')};
+  }
 
   @media (max-width: ${props => props.theme.breakpoints.small}px) {
     padding-left: 0;
   }
 `;
 
+export const AdditionalContentWrapper = styled.div`
+  flex: 1;
+  justify-content: flex-end;
+  display: flex;
+  * {
+    display: table-cell;
+    margin: 0;
+    margin-left: ${props => props.theme.spaceScale.spacing03};
+  }
```

Comment:
Why is this needed? It seems like I can remove these styles and it still works the same

Reply:
If you look at the Storybook example with two badges, and also the button, it cleans up the left spacing and fixes an issue with left button padding.

Reply:
Fair enough, I'll remove that then!